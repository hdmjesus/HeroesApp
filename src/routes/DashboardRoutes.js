import React from "react";
import { Route, Switch, Redirect } from "react-router-dom";
import { NavBar } from "../Components/ui/NavBar";
import { DcScreen } from "../Screen/DcScreen";
import { HeroesScreen } from "../Screen/HeroesScreen";
import { MarvelScreen } from "../Screen/MarvelScreen";

export const DashboardRoutes = () => {
  return (
    <>
      <NavBar />
      <div>
        <Switch>
          <Route exact path="/marvel" component={MarvelScreen} />
          <Route exact path="/heroe/:heroid" component={HeroesScreen} />
          <Route exact path="/dc" component={DcScreen} />
          <Redirect to="/marvel" />
        </Switch>
      </div>
    </>
  );
};
