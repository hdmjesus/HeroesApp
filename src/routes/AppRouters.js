import React from "react";
import { BrowserRouter, Route, Switch } from "react-router-dom";

import { LoginScreen } from "../Screen/LoginScreen";
import { DashboardRoutes } from "./DashboardRoutes";

const App = () => {
  return (
    <BrowserRouter>
      <Switch>
        <Route exact path="/login" component={LoginScreen} />
        <Route path="/" component={DashboardRoutes} />
      </Switch>
    </BrowserRouter>
  );
};

export default App;
