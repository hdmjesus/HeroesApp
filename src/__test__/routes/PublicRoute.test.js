import { mount, shallow } from "enzyme";
import React from "react";
import { PublicRoute } from "../../routes/PublicRoute";
import { MemoryRouter } from "react-router";
describe("Pruebas en <PrivateRoute/>", () => {
  const props = { location: { pathname: "/login" } };

  test("should show  the component if is not authenticated and save in localstorage", () => {
    const wrapper = mount(
      <MemoryRouter>
        <PublicRoute
          exact
          path="/login"
          component={() => <h1>hello</h1>}
          isAuthenticated={false}
          {...props}
        />
      </MemoryRouter>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("h1").exists()).toBe(true);
  });

  test("should  block the component if is  authenticated", () => {
    Storage.prototype.setItem = jest.fn(() => {});

    const wrapper = mount(
      <MemoryRouter>
        <PublicRoute
          exact
          path="/"
          component={() => <h1>hello</h1>}
          isAuthenticated={true}
          {...props}
        />
      </MemoryRouter>
    );
    expect(wrapper.find("h1").exists()).toBe(false);
  });
});
