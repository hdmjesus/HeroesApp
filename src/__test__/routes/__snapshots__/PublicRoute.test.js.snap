// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrivateRoute/> should show  the component if is not authenticated and save in localstorage 1`] = `
<h1>
  hello
</h1>
`;
